Remove SpawnScan scheduler (-ss)

FrostTheFox opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
Working on getting rid of -ss.
## Description
<!--- Describe your changes in detail -->
Removing the scheduler and associated configuration options, in favor of speedscan.
## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
The speed scan scheduler is a lot more advanced, works properly (for the most part) and is being maintained. The spawnscan scheduler is old and has not been touched in a while, and I'm pretty sure it's completely broken at this point.
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
Have not yet tested myself, though everything should work in theory. Passes flake8 in Atom.
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Removes old functionality
## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--  NOTE: In order to check code style locally and avoid having your build rejected by Travis, -->
<!--  run the following commands before you commit: `flake8 .` and `npm run lint`. Fix any -->
<!--  issues they point out. Note also that flake's NOQA is disabled on Travis. -->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

Arengo commented:
While I see ``-ss`` as broken and unneeded at the moment, I do see use for an easy output of spawns without a direct database access. So ``--dump-spawnpoints`` into a file is not bad.

FrostTheFox commented:
@Arengo I actually couldn't even get --dump-spawnpoints to work on my instance originally which was why I removed it too, took a look at what I removed and realized it was depending on -ss having defined a file, readded it and patched it so it's not done stupidly :)

Arengo commented on runserver.py:
```diff
@@ -291,15 +291,13 @@ def main():
-            with open(args.spawnpoint_scanning, 'w+') as file:
-                log.info('Saving spawn points to %s', args.spawnpoint_scanning)
+        if args.dump_spawnpoints:
+            with open('spawnpoints.json', 'w+') as file:
```
Thanks for re-adding. When we already have ``--dump-spawnpoints`` flag, why not making it a string flag to specify the file name to dump to. So it's basically just moving the ``-ss`` arg type to that one.

FrostTheFox replied:
https://github.com/RocketMap/RocketMap/pull/2010/commits/6fd77c8c813e49bf64a76621622ec4be3aba8a47

myntath commented:
I still use SpawnScan for very quickly scanning all gyms over very large areas (100s of kilometers) using information dumped from the database. If SpwanScan is to go (and it probably is ready to be cut) I think we should prioritise merging #1957 first. Otherwise I think there will be a genuine loss of functionality for a small group of users.

FrostTheFox commented:
Well, that's functionality that SpawnScan is not intended for :(

Obihoernchen commented:
Regarding --dump-spawnpoints please have a look at https://github.com/RocketMap/RocketMap/pull/1948.
We should replace the existing method anyways imo.

SkOODaT commented:
im still using -ss on my main map works great as long as you have a way to geofence... and with Arengos new PR -ss could be useful again

FrostTheFox commented:
@SkOODaT You're not really getting the true usage of -ss, SS has not actually worked for a while now because of the timer change from Niantic.

SkOODaT commented:
yea -speed is better in process of converting over now but generally i haven't had any issues running -ss timers pop up right at 30 mins good enough for ppl to catch things lol it was the fact i needed -ss for a geofence tool i had, (odd scanning area)..... unless theirs been changes since the version i run and the newest stuff  that messed up -ss , with Arengos new PR for geofence tho i guess -ss is really getting obsolete

FrostTheFox commented:
The timers might show 30 minutes, that doesn't mean they're really 30 minutes.

SkOODaT commented:
they are lol ive timed multiple ones to make sure map is live now 30 plus ppl on it if thier were timer issues id know lol

FrostTheFox commented:
Did you export spawnpoint data with timers? Because if so, that can make -ss work sort of, but that's not because it's -ss, that's because you already had the timers.

FrostTheFox commented:
Either way, -ss has no use even with any future PRs. -speed was designed to replace -ss and it has done so.

SkOODaT commented:
yea i exported and compressed the good old way like back in the day lol :)

bluemode commented:
Plz don't remove -ss! It is outdated and pretty much useless! That is exactly what we need to keep finding the pokemon. Geez

richardfoltin commented:
I'm running a map  that finds 50.000 pokemon/hour and using a customized -ss.

**Customizations:**
- ss gets spawnpoints from spawnpoint table that was found with -speed before
- spawnpoints get clustered on the fly (at start of each instance)
- the timetreshold of clustering is added as an argument
- added another speedlimiter (to work with -ss)

**Benefits:**
- much less processor/memory/hard disk usage compared to -speed
- needs less RMP because of clustering
- less workers needed -> less proxies needed
- I can cover a much bigger area because I can use -st 4 beehives (The pokemon are very dense at some areas. Without clustering -st 4 could not be scanned without skipping under the speed limit)
